extension/content/components/pages/RecipeFormPage.js

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+
+import { RecipeFormProvider } from "devtools/contexts/recipeForm";
+import {
+  useSelectedExperimenterEnvironmentAPI,
+  useSelectedNormandyEnvironmentAPI,
+} from "devtools/contexts/environment";
+import RecipeForm from "devtools/components/recipes/form/RecipeForm";
+
+export default function RecipeFormPage() {
+  const { recipeId, experimenterSlug } = useParams();
+  const [data, setData] = React.useState({});
+  const [importInstructions, setImportInstructions] = React.useState("");
+
+  const normandyApi = useSelectedNormandyEnvironmentAPI();
+  const experimenterApi = useSelectedExperimenterEnvironmentAPI();
+
+  React.useEffect(() => {
+    if (recipeId) {
+      normandyApi.fetchRecipe(recipeId).then((recipeData) => {
+        setData(recipeData.latest_revision);
+        setImportInstructions("");
+      });
+    } else if (experimenterSlug) {
+      experimenterApi
+        .fetchRecipe(experimenterSlug)
+        .then(({ comment, ...recipeData }) => {
+          setData(recipeData);
+          setImportInstructions(comment);
+        });
+    } else {
+      setData({});
```

Comment:
was fiddling with it and the create workflow wasnt working for me until I inserted filter_object and arguments here...
not exactly sure if this is the right place to do that though...